packages/bundlers/experimental/src/ExperimentalBundler.js

```diff
@@ -534,8 +534,16 @@ function createIdealGraph(
                     : referencingBundle.needsStableName,
               });
               bundleId = bundleGraph.addNode(bundle);
+
               // Store Type-Change bundles for later since we need to know ALL bundlegroups they are part of to reduce/combine them
-              typeChangeIds.add(bundleId);
+              if (parentAsset.type !== childAsset.type) {
+                typeChangeIds.add(bundleId);
```

Comment:
I think we should not merge parallel/lazy bundles, because the transformer that created those dependencies might not expect it. For example, if you have two `<script>` tags in an HTML page, those have parallel priority, but merging them would result in both scripts pointing to the same URL.